packages/next/src/client/legacy/image.tsx

```diff
@@ -156,8 +156,10 @@ function defaultLoader({
         process.env.NEXT_RUNTIME !== 'edge'
       ) {
         // We use dynamic require because this should only error in development
-        const { hasMatch } = require('../../shared/lib/match-remote-pattern')
-        if (!hasMatch(config.domains, config.remotePatterns, parsedSrc)) {
+        const {
+          hasRemoteMatch,
+        } = require('../../shared/lib/match-remote-pattern')
+        if (!hasRemoteMatch(config.domains, config.remotePatterns, parsedSrc)) {
```

Comment:
should this check be present for `localPatterns` as well?

Reply:
Good catch! I had it on the new image component but not the legacy image component.

Added in [246d0c2](https://github.com/vercel/next.js/pull/70529/commits/246d0c2bd9c504a3492005192c1eb7d295fc97a4)